### ISSUE TITLE
Dependencies not automatically install in docker image while debugging locally

### ISSUE THREAD

**Issue by LongNTBui** [verzac/typescript-aws-sam#5]
Thanks for this useful repo.

During local debugging by running the following command:
 - sam local invoke HelloWorldFunction --debug-port 5858

I always get an error message saying that "axios" can not be found. This seems to be an issue of the dependencies in package.json file does not get installed inside the docker image.

Is it possible to change some config in template.yml file to make it happen? Thanks

Note:
I understand that if I copy the dependencies to "dist" folder it works. I think if I could tell "sam" to reference the node_modules folder inside template.yml file it would be more efficient.

**Comment by verzac** [comment]
Hi there LongNTBui!

TL:DR; no it can't.

Unfortunately, this is the drawback of AWS SAM. It doesn't make any assumptions on how you should install your dependencies and therefore it doesn't try to install it for you (which is fair, because it supports other languages too and there are _usually_ multiple ways you can install dependencies in these languages; e.g. Node has yarn and NPM).

The way SAM works is that it just registers the folder(s) specified in the `CodeUri` section part of your Lambda functions in `template.yaml` (in this case `./dist`), and loads the folder onto the Docker container and executes it based on the information you give it in `template.yaml`. That is it: nothing less, and nothing more.

I agree, it's a bit annoying, and it used to annoy and confuse the hell out of me when errors such as "axios can not be found" pops up. I understand that this is even more annoying when you have to precompile your stuff first (as in the case when you're using TS).

What I recommend is to include the "dependency installation part" as part of your start script, and this is why the "dependency installation" part is included as part of the "npm run start-api" script (i.e. `cpx "./node_modules/**/*" ./dist`).

If you want to see how I do it in my latest project, see https://github.com/verzac/news-neutrality-scraper. It's what I'm most comfortable with so far. I may start looking at using webpack soon, but my setup in that project has been satisfactory for me, so I can't be bothered with changing it.

Feel free to ask any more questions :)

**Comment by verzac** [comment]
Closing this due to a lack of activity. Feel free to reopen or to comment on this issue if you still have any unanswered questions :)
